pom.xml

```diff
@@ -164,6 +164,7 @@
     <commons.collections.version>3.2.2</commons.collections.version>
     <scala.version>2.12.10</scala.version>
     <scala.binary.version>2.12</scala.binary.version>
+    <scalac.arg.unused-imports>-Ywarn-unused-import</scalac.arg.unused-imports>
```

Comment:
Shall we just move it to `SparkBuild.scala` and let Maven doesn't care about that?

Reply:
There should be only some warnings when maven build now, seems that @MaxGekk wants maven check it as error too?

Reply:
Is my guess right? @MaxGekk

Reply:
If it's easy to make both Maven and SBT build throw an error, it's fine. If that's difficult, let's move it to `SparkBuild.scala`, and make it SBT specific.

Reply:
> Shall we just move it to SparkBuild.scala and let Maven doesn't care about that?

It may be useful for Maven, it can also help Maven users to check this although it is only warnings with maven build

Reply:
> If it's easy to make both Maven and SBT build throw an error, it's fine. If that's difficult, let's move it to SparkBuild.scala, and make it SBT specific.

OK ~ I can add a followup to do this change, I'm not sure how Maven users can self check before submitting code ....

Reply:
Actually maven users will not fail to compile at present, just some warnings about `unused-imports` print to console now...

Reply:
Yeah, that's fine. We can make SBT side complete alone first, and think about Maven side separately later. Maven side will be secondary for now because PR builder can check anyway.

Reply:
OK~ do it later, a little busy

Reply:
Sure thanks :-)

Reply:
@LuciferYang, just checking. do we have a JIRA ticket for Maven side to follow up? For completeness, it's good to do it for Maven as well anyway.

Reply:
@HyukjinKwon SPARK-33560 added :)

Reply:
Thanks @LuciferYang

Reply:
@HyukjinKwon Is it necessary for us to add more compiler checking further? like unused-locals?

Reply:
We can give a shot. Are they a lot of instances of `unused-locals` to fix? If there are too many, I think we should collect some more feedback from other committers because it makes more difficult to maintain the codes (e.g., reverting and backporting).

Reply:
OK ~ I'll collect the details of the issue like `unused-locals` first, then file a new Jira for tracking and  discussion. thanks @HyukjinKwon